src/index.js

```diff
@@ -2,66 +2,56 @@ let listenAction
 
 export default {
   bind(el, binding) {
-    const params = binding.value || {}
+    const params = binding.value || {},
+      stickyTop = params.stickyTop || 0,
+      zIndex = params.zIndex || 1000,
+      elStyle = el.style
 
-    const stickyTop = params.stickyTop || 0
-    const zIndex = params.zIndex || 1000
-
-    el.style.position = '-webkit-sticky'
-    el.style.position = 'sticky'
+    elStyle.position = '-webkit-sticky'
+    elStyle.position = 'sticky'
 
     // if the browser support css sticky（Currently Safari, Firefox and Chrome Canary）
-    if (~el.style.position.indexOf('sticky')) {
-      el.style.top = `${stickyTop}px`
-      el.style.zIndex = zIndex
+    if (~elStyle.position.indexOf('sticky')) {
+      elStyle.top = `${stickyTop}px`
+      elStyle.zIndex = zIndex
       return
     }
 
-    const elementChild = el.firstElementChild
-    elementChild.style.cssText = `left: 0; right: 0; top: ${stickyTop}px; index: ${zIndex}`
+    const elementChild = el.firstElementChild.style
+    elementChild.cssText = `left: 0; right: 0; top: ${stickyTop}px; index: ${zIndex}`
 
     let active = false
 
     const sticky = () => {
       if (active) {
         return
       }
-      if (!el.style.height) {
-        el.style.height = el.offsetHeight + 'px'
+      if (!elStyle.height) {
+        elStyle.height = `${el.offsetHeight}px`
       }
-      elementChild.style.position = 'fixed'
+      elementChild.position = 'fixed'
       active = true
     }
 
     const reset = () => {
       if (!active) {
         return
       }
-      elementChild.style.position = ''
+      elementChild.position = ''
       active = false
     }
 
     const check = () => {
       const offsetTop = el.getBoundingClientRect().top
       if (offsetTop <= stickyTop) {
         sticky()
-      } else {
-        reset()
+        return
       }
+      reset()
     }
 
-    let scrollerTimer // for bad user experience scroll in mobile
-    let scrollEndTimer // for clear scrollerTimer when scroll end
     listenAction = () => {
-      clearTimeout(scrollEndTimer)
-      scrollEndTimer = setTimeout(() => {
-        clearInterval(scrollerTimer)
-        scrollerTimer = null
-      }, 1000)
-
-      if (!scrollerTimer) {
-        scrollerTimer = setInterval(check, 30)
-      }
+      setTimeout(check, 300);
```

Comment:
That's why I proposed this change. The event itself will execute the check function on each scroll. The current code do a lot and it Does delay the check function (line 62).

Why do I delay it 300 milliseconds? When the user fast scroll it might not get the scroll position correctly. If I delay it, it will solve that issue. You do the same thing I did, except you're using a setInterval. 

setInterval is not needed because we are listening to an event. An event by itself remembers a setInterval, even though it not. When using setInterval, you need to clean it up with clearInterval. In my case, your code runs only once with the same delay you are using.

Run my PR and check it out.